lua/lspsaga/rename.lua

```diff
@@ -86,6 +86,41 @@ local rename = function()
   apply_action_keys()
 end
 
+--- WorkspaceEdit -> Location[]
+---@param changes table
+---@return table
+local workspaceedit_changes_to_location_list = function(changes)
+  local result = {}
+
+  for uri, text_edits in pairs(changes) do
+    for _, text_edit in ipairs(text_edits) do
+      table.insert(result, {
+        uri = uri,
+        range = text_edit.range
+      })
+    end
+  end
+
+  return result
+end
+
+local rename_handler = function(_, result, ctx, _)
+  -- source: vim.lsp.handlers['textDocument/rename']
+  if not result then return end
+  local client = vim.lsp.get_client_by_id(ctx.client_id)
+  util.apply_workspace_edit(result, client.offset_encoding)
+
+  if config.rename_output_qflist then
+    local offset_encoding = vim.lsp.get_client_by_id(ctx.client_id).offset_encoding
+    local locations = workspaceedit_changes_to_location_list(result.changes)
+    if #locations == 0 then
+      return
+    else
+      vim.fn.setqflist(vim.lsp.util.locations_to_items(locations, offset_encoding))
```

Comment:
merged https://github.com/nvim-telescope/telescope.nvim/pull/1742 few hours ago and the author did great job explaining the how `r` mode enables stacking lists and using other qf functions to navigate between them. Maybe specify mode as last argument? `:h setqflist-action`

Reply:
Does that mean that I should specify `' '`?
According to the help, if you don't specify an action, it will be added to the list of existing quick fixes.
However, there are cases where it is better to explicitly specify `' '`.
Should I modify it?

Reply:
Yes, should be specified as 'r' so that when you have say a quickfix with a number of items and you open another list,your list doesn't get lost and instead you can navigate back to the old list with :colder

Reply:
Yes, my bad, `' '` sorry for the misunderstanding 😆.

Reply:
Don't worry about it :+1: